libs/remote_storage/src/s3_bucket.rs

```diff
@@ -707,52 +711,51 @@ impl RemoteStorage for S3Bucket {
                 }
                 break;
             }
+            // Limit the number of versions deletions, mostly so that we don't
+            // keep requesting forever if the list is too long, as we'd put the
+            // list in RAM.
+            // Building a list of 100k entries that reaches the limit roughly takes
+            // 40 seconds, and roughly corresponds to tenants of 2 TiB resident_size.
```

Comment:
```suggestion
            // 40 seconds, and roughly corresponds to tenants of 2 TiB physical size.
```

Resident size has no meaning in this context, it's subject to the eviction changes etc.